### PR TITLE
Add is_subset methods for BitList and BitVector

### PR DESCRIPTION
**Pull request by pawanjay176**
Adds `is_subset` methods that would be useful in https://github.com/sigp/lighthouse/pull/4327

### CODE REVIEW

File: src/bitfield.rs [hunk 1]
```diff
@@ -765,6 +798,30 @@ mod bitvector {
...
+        let c = BitVector16::from_raw_bytes(smallvec![0b1100, 0b1001], 16).unwrap();
+
+        assert_eq!(a.len(), 16);
+        assert_eq!(b.len(), 16);
+        assert_eq!(c.len(), 16);
+
+        // a vector is always a subset of itself
+        assert!(a.is_subset(&a));
+        assert!(b.is_subset(&b));
+        assert!(b.is_subset(&c));
```
Comment (divagant-martian) [inline src/bitfield.rs:814]:
```suggestion
        assert!(c.is_subset(&c));
```

File: src/bitfield.rs [hunk 2]
```diff
@@ -235,6 +236,32 @@ impl<N: Unsigned + Clone> Bitfield<Variable<N>> {
...
+            // Here, `a` is a subset of `b` but if we handle it similarly to the `Ordering::Equal` and `Ordering::Less` case
+            // we get `a.intersection(b) = 0b1011 0b0010 ≠ a` and `a.is_subset(&b)` would return `false`.
+            Ordering::Greater => {
+                let mut result =
+                    Self::with_capacity(self.len()).expect("max len always less than N");
+                for i in 0..other.bytes.len() {
+                    result.bytes[i] = other.bytes.get(i).copied().unwrap_or(0);
+                }
+                let intersection = self.intersection(&result);
+                intersection == *self
```
Comment (divagant-martian) [inline src/bitfield.rs:261]: can this maybe be written as `self.difference(other).is_zero()`?

File: src/bitfield.rs [hunk 3]
```diff
@@ -1209,6 +1266,46 @@ mod bitlist {
...
+        let c = BitList1024::from_raw_bytes(smallvec![0b1100, 0b1001], 16).unwrap();
+
+        assert_eq!(a.len(), 16);
+        assert_eq!(b.len(), 16);
+        assert_eq!(c.len(), 16);
+
+        // a vector is always a subset of itself
+        assert!(a.is_subset(&a));
+        assert!(b.is_subset(&b));
+        assert!(b.is_subset(&c));
```
Comment (divagant-martian) [inline src/bitfield.rs:1282]:
```suggestion
        assert!(c.is_subset(&c));
```

File: src/bitfield.rs (hunk 2 above)
Comment (pawanjay176) [reply src/bitfield.rs:261]: Yep, this is much cleaner and would work in all cases I think. 
Thank you!

**Review by michaelsproul — APPROVED** [review]
Looks good, thanks!

I'll cut a new release

**Comment by michaelsproul** [comment]
New release is out: https://crates.io/crates/ssz_types

**Pull request merged**